vespalib/src/vespa/vespalib/datastore/buffer_type.h

```diff
@@ -2,11 +2,12 @@
 
 #pragma once
 
-#include <cstdint>
-#include <cstddef>
+#include "atomic_entry_ref.h"
+#include <string>
 
 namespace vespalib::datastore {
 
+using ElemCount = uint32_t;
```

Comment:
Wrong type for ElemCount. A buffer can contain more than 4G elements. E.g. when using 1024 buffers and array size > 1024. Clamping `_maxArrays`, `_minArrays` and `_numArraysForNewBuffer` in `BufferTypeBase` might work.